pkg/apis/contrail/v1alpha1/control_types.go

```diff
@@ -263,41 +274,45 @@ func (c *Control) InstanceConfiguration(request reconcile.Request,
 			RabbitmqVhost:       rabbitmqSecretVhost,
 			CAFilePath:          certificates.SignerCAFilepath,
 		})
-		data["dns."+podList.Items[idx].Status.PodIP] = controlDNSConfigBuffer.String()
+		data["dns."+podIP] = controlDNSConfigBuffer.String()
 
 		var controlNodemanagerBuffer bytes.Buffer
 		configtemplates.ControlNodemanagerConfig.Execute(&controlNodemanagerBuffer, struct {
 			ListenAddress       string
+			DataIP              string
 			CollectorServerList string
 			CassandraPort       string
 			CassandraJmxPort    string
 			CAFilePath          string
 		}{
-			ListenAddress:       podList.Items[idx].Status.PodIP,
+			ListenAddress:       podIP,
+			DataIP:              dataIP,
 			CollectorServerList: configCollectorEndpointListSpaceSeparated,
 			CassandraPort:       cassandraNodesInformation.CQLPort,
 			CassandraJmxPort:    cassandraNodesInformation.JMXPort,
 			CAFilePath:          certificates.SignerCAFilepath,
 		})
-		data["nodemanager."+podList.Items[idx].Status.PodIP] = controlNodemanagerBuffer.String()
+		data["nodemanager."+podIP] = controlNodemanagerBuffer.String()
 
 		var controlProvisionBuffer bytes.Buffer
 		configtemplates.ControlProvisionConfig.Execute(&controlProvisionBuffer, struct {
 			ListenAddress string
+			DataIP        string
 			APIServerList string
 			ASNNumber     string
 			BGPPort       string
 			APIServerPort string
 			Hostname      string
 		}{
-			ListenAddress: podList.Items[idx].Status.PodIP,
+			ListenAddress: podIP,
```

Comment:
Is now "ListenAddress" field used?
Why not just use this ListenAddress and assign dataIP there.
`ListenAddress: dataIP`

Technically it is listen address of this service am I right?